variables.tf

```diff
@@ -21,21 +44,49 @@ variable "description" {
   description = "The Security Group description."
 }
 
-variable "id" {
-  type        = string
-  default     = ""
-  description = <<-EOT
-    The external Security Group ID to which Security Group rules will be assigned.
-    Required to set `security_group_enabled` to `false`.
-  EOT
-}
-
 variable "rules" {
   type        = list(any)
-  default     = null
+  default     = []
   description = <<-EOT
-    A list of maps of Security Group rules. 
-    The values of map is fully complated with `aws_security_group_rule` resource. 
+    A list of maps of Security Group rules.
+    The keys and values of the maps are fully compatible with the `aws_security_group_rule` resource, except
+    for `security_group_id` which will be ignored.
     To get more info see https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule .
   EOT
 }
+
+variable "open_egress_enabled" {
+  type        = bool
+  default     = false
+  description = <<-EOT
+    A convenience. Add to the rules in `var.rules` a rule that allows all egress.
+    If this is false and `var.rules` does not specify any egress rules, then
+    no egress will be allowed.
+    EOT
+}
+
+variable "rule_matrix" {
+  type        = any
+  default     = { rules = [] }
+  description = <<-EOT
+    A convenience. Apply the same list of rules to all the provided security groups and CIDRs and self.
+    Type is object as specified in the default, but keys are optional except for `rules`.
+    The `rules` list is a list of maps that are fully compatible with the `aws_security_group_rule` resource,
+    but any keys already at the top level will be ignored.  Rules keys listed in the default are required, except for `description`.
```

Comment:
"but any keys already at the top level will be ignored" - I think we need to add more details on what is "the top level"

Reply:
> "but any keys already at the top level will be ignored" - I think we need to add more details on what is "the top level"

I agree and am open to suggestions. The issue here is that I cannot document the schema in the `type` statement as normal because we want to make the input more flexible than Terraform allows. I think the documentation is already too long for the standard documentation block and yet I think it needs even more. We can put fuller documentation in README, so maybe just point to that?